qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py

```diff
@@ -74,11 +76,132 @@ def get_raw(self) -> np.ndarray:
                                                 self.steps,
                                                 self.mode)
         else:
-            raise RuntimeError("No instrument attached to Parameter")
+            raise RuntimeError("No instrument attached to Parameter.")
 
         return data
 
 
+class TimeTrace(ParameterWithSetpoints):
+    """
+    A parameter class that holds the data corresponding to the time dependence of
+    current and voltage.
+    """
+
+    def __init__(self, name: str, instrument: Instrument, **kwargs) -> None:
+
+        super().__init__(name=name,
+                         instrument=instrument,
+                         **kwargs)
+
+    def _validateTimeTrace(self, mode: str) -> None:
+        """
+        A helper function that validates the timetrace mode is correct
+        as well as compares the integration time with measurement interval
+        for accurate results.
+
+        Args:
+            mode: User defined mode for the timetrace. It can be either
+            current, "i", or voltage "v".
+
+        Raises:
+            ValueError: If the timetrace mode is not set to either curent or
+            voltage.
+        """
+
+        if mode not in ['i', 'v']:
```

Comment:
Since the `timetrace_mode` parameter already has a validator, there is no reason to re-validate here (or anywhere else) as long as that parameter's value is used as `mode`.